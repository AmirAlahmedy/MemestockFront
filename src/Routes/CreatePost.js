import React, { Component } from 'react';
import './CreatePost.css';

class CreatePost extends Component {
    
}

export default CreatePost;