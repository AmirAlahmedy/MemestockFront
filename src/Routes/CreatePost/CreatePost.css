    .createPostContainer{
        font-style: inherit;
        color:#333;
        width:50%;
        max-width: 600px;
        min-width: 300px;
        overflow: hidden;
        margin: 0 auto;
        border-radius: 4px;
        background: #fff;
        padding: 1em;
        box-shadow: 0 0 7px rgba(51,51,51,0.5);        
    }
    h3{
        font-style: inherit;
        font-size:20px;
        font-weight: 500;
        float:center;       
        margin:0;
        padding-top:10px;
        padding-left:1%;
        border:0;
    }
    .formGroupThreadCheckbox{
        font-size:18px;
        text-align: left;
        padding-bottom:15px;
    }
    .formGroupThreadComponent{
        font-size:18px;
        text-align: left;
        padding: 1em;
        display: block;
        box-sizing: border-box;
    }
    .formGroupThreadComponent textarea{
        margin: 0 auto;
        border-radius:4px;
        border: 1px solid #ccc;
        resize: none;
        overflow-y: auto;
        overflow-x: hidden;
        font-size: 0.85em;
        padding: 1em;
        width: 100%;
        font-family: inherit;
    }
    .spoilerLabel{
        margin: 0 1em;
    }
    .formGroupThreadComponent input{
        border-radius:4px;
        font-family: inherit;
        border: 1px solid #ccc;
        padding: 1em;
        font-size: 0.85em;
        width:100%;
    }
    .formGroupThreadComponent label {
        padding-bottom: 0.7em;
        display: inline-block;
    }
    .threadPageCreateButton {
        background-color: #0079D3;
        border: none;
        color: #fff;
        outline: none;
        cursor: pointer;
        /* font: inherit; */
        font-size: 14px;
        padding: 10px;
        font-weight: bold;
        border-radius: 3px;
        width: calc(100% - 2em);
        padding-left: 0;
        letter-spacing: 0.5px;
        line-height: 32px;
        text-transform: uppercase;
        margin: 0 1em;
    }
    .threadPageCreateButton:hover{
        background-color: #33a8ff;
    }
.errorMessageCreatePost{
    font-style: inherit;
    color:maroon;
    font-size: 0.77em;
}
.threadPhotoInput{
    display: none;
}
.formGroupThreadComponent .threadPageCreateButton{
    width: 20%;
    min-width: 250px;
    text-align: center;
}
.imageContainer{
    display: flex;
    align-items: center;
}
.uploadedImage{
    width: 250px;
    height: 250px;
    background-position: center;
    background-size: cover;
    background-repeat: no-repeat;
}
/* *:focus,
*:active{
    border-color: #2684FF !important;
    outline: none;
} */