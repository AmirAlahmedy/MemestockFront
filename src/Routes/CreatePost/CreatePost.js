import React, { Component } from 'react';
import './CreatePost.css';

//import { Button, Form, FormGroup, Label, Input, FormText } from 'reactstrap';

import axios from '../../axios-orders';

class CreatePost extends Component {
  //assume all submitted for now
  constructor(props) {
    super(props);
    this.state = {

      title: '',
      body: '',
      spoiler: false,
      subreddit: '',
      image: null,
      error: false,
      errornumber: 0
    }

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleFileChange = this.handleFileChange.bind(this);
  };



  handleChange(e) {
    this.setState({
      spoiler: e.target.checked
    });
  }

  /**
     * For sending a creation post request to the backend 
     * @function handleSubmit
     * @param {event} - onClick event 
     */

  handleSubmit(e) {
    this.setState({
      error: false,
      errornumber: 0
    })
    e.preventDefault();
    var srdata = {
      title: document.getElementById("threadPageTitleField").value,
      threadBody: document.getElementById("threadPageBodyField").value,
      postFile: {},
      spoiler: this.state.spoiler,
      base64image: this.state.image
    }
    let checker = "";

    if (document.getElementById("threadPageSubredditNameField").value === checker) {
      //alert("Please provide an existing Subreddit name!");
      this.setState({
        error: true,
        errornumber: 1
      })
      return;
    }
    else if (document.getElementById("threadPageTitleField").value === checker) {
      this.setState({
        error: true,
        errornumber: 2
      })
      //alert("Please provide a Thread Title!");
      return;
    }
    else if (document.getElementById("threadPageBodyField").value === checker) {
      //alert("Please provide a Thread Body!");
      this.setState({
        error: true,
        errornumber: 3
      })
      return;
    }

    let subredditname = document.getElementById("threadPageSubredditNameField").value;
    let headers = {
      auth: localStorage.getItem("token")
    }
    axios.post('/sr/' + subredditname + '/thread', srdata, { "headers": headers })
      .then(res => {
        console.log(res);
        console.log(res.status);
        window.location.href = "/thread/" + res.data._id + "?srName=" + subredditname;
      })
      .catch(error => {
        alert(error.response);
        this.setState({
          error: true
        })
      });
  }
  componentDidMount() {
    document.body.classList.add("verticalAlign");
  }
  componentWillUnmount() {
    document.body.classList.remove("verticalAlign");
  }

  handleFileChange(e) {
    const files = e.target.files;
    if (!files || !files.length) return;
    const reader = new FileReader();

    reader.onload = (evtReader) => {
      if (!evtReader.target.result) return;
      this.setState({
        image: evtReader.target.result
      });
    }
    reader.readAsDataURL(files[0]);
  }
  render() {
    return (
      <div className="createPostContainer">
        <h3>CREATE A POST</h3>

        <hr></hr>
        <form onSubmit={this.handleSubmit}>
          <div className="formGroupThreadComponent">
            <label for="SubredditName">Subreddit Name</label>
            <br></br>
            <input type="text" name="text" id="threadPageSubredditNameField" placeholder="Enter Existing Subreddit Name" onChange={this.handleChange} value={this.state.value} />
          </div>
          {
            this.state.error ?
              <div>
                {
                  this.state.errornumber == 1 ?
                    <div className="errorMessageCreatePost">
                      *Please provide an existing Subreddit name!
            </div> : <div></div>
                }
              </div> : <div></div>
          }
          <div className="formGroupThreadComponent">
            <label for="ThreadTitle">Thread Title</label>
            <textarea type="textarea" name="text" id="threadPageTitleField" placeholder="Enter Title Here" onChange={this.handleChange} value={this.state.value} />
          </div>
          {
            this.state.error ?
              <div>
                {
                  this.state.errornumber == 2 ?
                    <div className="errorMessageCreatePost">
                      *Please provide a Thread Title!
            </div> : <div></div>
                }
              </div> : <div></div>
          }
          <div className="formGroupThreadComponent">
            <label for="ThreadBody">Thread Body</label>
            <textarea type="textarea" name="text" id="threadPageBodyField" placeholder="Enter Body Here" onChange={this.handleChange} value={this.state.value} />
          </div>
          {
            this.state.error ?
              <div>
                {
                  this.state.errornumber == 3 ?
                    <div className="errorMessageCreatePost">
                      *Please provide a Thread Body!
            </div> : <div></div>
                }
              </div> : <div></div>
          }

          <div className="formGroupThreadComponent">
            <label for="coverPhoto">Cover Photo:</label>
            <input type="file" name="coverPhoto" id="coverPhoto" onChange={this.handleFileChange} />
          </div>
          <div className="formGroupThreadCheckbox">
            <label className="spoilerLabel" for="Spoiler">Is it a spoiler?</label>
            <input type="checkbox" name="Spoiler" id="Spoiler" onChange={this.handleChange} value={this.state.spoiler} />
          </div>
          <button className="threadPageCreateButton">CREATE</button>
        </form>
      </div >

    )
  } 
}
export default CreatePost;
