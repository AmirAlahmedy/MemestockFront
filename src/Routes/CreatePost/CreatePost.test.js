import React from "react";
import { create } from "react-test-renderer";
import CreatePost from "./CreatePost";


describe("CreatePost component snapshot", () => {
    test("it matches the snapshot", () => {
      const component = create(<CreatePost />);
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
  
