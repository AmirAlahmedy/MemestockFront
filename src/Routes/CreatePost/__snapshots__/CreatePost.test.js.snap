// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePost component snapshot it matches the snapshot 1`] = `
<div
  className="createPostContainer"
>
  <h3>
    CREATE A POST
  </h3>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <div
      className="formGroupThreadComponent"
    >
      <label
        for="SubredditName"
      >
        Subreddit Name
      </label>
      <br />
      <input
        id="threadPageSubredditNameField"
        name="text"
        onChange={[Function]}
        placeholder="Enter Existing Subreddit Name"
        type="text"
      />
    </div>
    <div />
    <div
      className="formGroupThreadComponent"
    >
      <label
        for="ThreadTitle"
      >
        Thread Title
      </label>
      <textarea
        id="threadPageTitleField"
        name="text"
        onChange={[Function]}
        placeholder="Enter Title Here"
        type="textarea"
      />
    </div>
    <div />
    <div
      className="formGroupThreadComponent"
    >
      <label
        for="ThreadBody"
      >
        Thread Body
      </label>
      <textarea
        id="threadPageBodyField"
        name="text"
        onChange={[Function]}
        placeholder="Enter Body Here"
        type="textarea"
      />
    </div>
    <div />
    <div
      className="formGroupThreadCheckbox"
    >
      <label
        className="spoilerLabel"
        for="Spoiler"
      >
        Is it a spoiler?
      </label>
      <input
        id="Spoiler"
        name="Spoiler"
        onChange={[Function]}
        type="checkbox"
        value={false}
      />
    </div>
    <button
      className="threadPageCreateButton"
    >
      CREATE
    </button>
  </form>
</div>
`;
